metaflow/plugins/kfp/kfp.py

```diff
@@ -564,6 +565,34 @@ def _step_cli(self, node: DAGNode, task_id: str, user_code_retries: int) -> str:
         cmds.append(" ".join(entrypoint + top_level + step))
         return " && ".join(cmds)
 
+    @staticmethod
+    def _create_resource_based_node_type_toleration(
+        cpu: float, memory: float
+    ) -> Optional[V1Toleration]:
+        """Allow large enough pod to use higher cost nodes
+
+        The following node types are considered for setting the threshold:
+        c5.4xlarge: 16 vCPU, 32 GB
+        r5.12xlarge: 48 vCPU, 384 GB
+        Resource threshold are lower than machine resource boundary to take overheads into
+        account.
+
+        Toleration allows pods to utilize larger nodes without enforcement.
+        Setting a low threshold for using larger host allow more freedom for scheduler
+        and potentially higher utilization rate.
+        """
+        # No need to validate value - already done by set_<resource>_request above
+
+        if cpu >= 12 or memory >= 24:  # 80% of hard limit - c5.4xlarge: 16 vCPU, 32 GB
```

Comment:
now: how many resources does the system use? Can we be a bit more aggressive?

```
16-12 = 4 cpus?
32 - 24 = 8 GB?
```

Maybe 
`if cpu >= 14 or memory >= 28:`

Reply:
Moving the conversation to slack

Reply:
Updated threshold based on resource deep dive in https://docs.google.com/document/d/1dKkQke78gqjAV2Ww_JnEBipHy9zs2NXCQHailhFiOIs/edit#

Summary of available resource investigation added to comments.